x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -253,6 +253,7 @@ public synchronized void stop() {
         }
 
         IndexerState state = getIndexer().stop();
+        stateReason.set(null);
```

Comment:
This was actually a bug. The state reason could still stick around even after the indexer stops. We should make sure that we clear out the reason on `stop` as well as `start`

Reply:
What if the transform fails then is force stopped, should we not preserve the failure reason in that case?

Reply:
I don't think we should.  The whole point of keeping the persistent task around when the transform is in the `failed` state is to remember stuff about the failure, like which node it was on and the reason.  By force-stopping a failed transform the user is effectively saying "I've investigated this failure as much as I'll ever want to, now I'm ready to move on".